src/db/migrate.rs

```diff
@@ -364,6 +364,22 @@ pub fn migrate(version: Option<Version>, conn: &Connection) -> CratesfyiResult<(
             "DROP TABLE compression_rels;
              ALTER TABLE files DROP COLUMN compression;"
         ),
+        migration!(
+            context,
+            // version
+            15,
+            // description
+            "Fix owner_rels.cid foreign key reference",
+            // upgrade query
+            "
+            ALTER TABLE owner_rels DROP CONSTRAINT owner_rels_cid_fkey;
+            ALTER TABLE owner_rels ADD FOREIGN KEY (cid) REFERENCES crates(id);
```

Comment:
So if I understand this correctly:

- The code uses `cid` to mean `crate_id` everywhere.
- The database foreign key referenced the `releases.id` column, which didn't match the usage.
- The reason we didn't catch it until now is because both are strictly increasing, and there are always more releases than crates. So the foreign constraint held even though the semantic meaning was incorrect.
- This changes the foreign key constraint but nothing else, because all the joins are already done on `owners INNER JOIN crates ON owners.cid = crates.id`, which is the correct behavior.

However I don't think the last point is entirely true: When I wrote `delete_crate` I thought this was actually the release id. So I think that `delete_crate` is actually very broken right now and needs to be fixed as part of this change: https://github.com/rust-lang/docs.rs/blob/7ab83067e3519e30d3ca3214c49b8bbddceca34c/src/db/delete.rs#L65 and https://github.com/rust-lang/docs.rs/blob/7ab83067e3519e30d3ca3214c49b8bbddceca34c/src/db/delete.rs#L102

cc @pietroalbini: **do not delete any more crates** until this is fixed.

Reply:
Yep, that all sounds correct, I've fixed it in the latest commit. I don't see any easy way to really test the behaviour.

Reply:
Fortunately this part of the code is (since #917) self-correcting so the worst that can happen is that the owner is missing until the next publish. Thanks for working on this :)